__tests__/api.test.js

```diff
@@ -89,7 +89,62 @@ describe('GET /api/articles/:article_id', () => {
         });
     })
 })
-
+describe('POST /api/articles/:article_id/comments', () => {
+    test("returns the new comment that has been created along with 201 status", () => {
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send({
+            username: "butter_bridge",
+            body: "well well well"
+        })
+        .expect(201)
+        .then(({body}) => {
+            const comment = body.comment[0]; 
+            expect(comment).toMatchObject({ 
+                body: "well well well",
+                votes: expect.any(Number),
+                article_id: 2,
+                author: "butter_bridge",
+                comment_id:  expect.any(Number)
+            });
+        });
+    })
+    test('if client doesnt supply required fields return status code 400 and a helpful error messge', () =>{
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send({
+        username: "butter_bridge"     
+        })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.message).toBe('both username and comment body is required')
+        })
+    })
+    test('if user name doesnt exist in database return error code 401 and helpful message', () => {
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send({
+        username: 'herbert_sherbert', 
+        body: 'hello great'
+        })
+        .expect(401)
+        .then(({ body }) => {
+            expect(body.message).toBe('user name does not exist')
+        })
+    })
+    test('if client sends incorrect data in either property sends a 400 error message and helpful message',() => {
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send({
+        username: 9, 
+        body: 9
+        })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.message).toBe('invalid data type')
+        })
+    })
+})
```

Comment:
Please include two more tests here, one for when the ID is invalid, and one where there's no article with the given ID.

Reply:
done